internal/pwru/utils.go

```diff
@@ -20,6 +21,14 @@ func GetFuncs() (Funcs, error) {
 
 	callback := func(typ btf.Type) {
 		fn := typ.(*btf.Func)
+
+		if pattern != "" {
+			match, _ := regexp.Match(pattern, []byte(fn.Name))
```

Comment:
I think we can optimize it by compiling the regexp first (with `regexp.Compile`), and then matching on the compiled regexp.